app/models/unavailable_service.rb

```diff
@@ -0,0 +1,21 @@
+class UnavailableService
+  attr_reader :country, :lgsl, :config
+
+  def initialize(lgsl, country, config)
+    raise StandardError, "Must supply an LGSL code" if lgsl.blank?
+    raise StandardError, "Must supply a valid config" unless config.is_a?(Hash)
+
+    @lgsl = lgsl.to_i
+    @country = country
+    @config = config
+    @service_path = ["services", lgsl]
+  end
+
+  def unavailable?
+    config.dig(*@service_path).present? && config.dig(*@service_path).include?(country)
+  end
+
+  def self.load_config
```

Comment:
Am I right in assuming a UnavailableService object represents a single service? Might be cleaner and easier to understand responsibilities, if we extract the information about all service from the object.

We could follow the pattern in the Brexit checker, where there is a class method that returns an array of objects. But each object is only aware of itself i.e. we don't store the entire config file for all object on each object. This would simplify things, as the config attribute is only for the particular service and you don't need to dig for it and you wouldn't need the derived service_path attribute.

https://github.com/alphagov/finder-frontend/blob/979b344ab47708a5e2519d5e7b16e80eac5b1f0f/app/lib/brexit_checker/action.rb#L46

Also unclear if this object is scoped per country? I don't think it needs to be if we made the `unavailable?` method accept a country name param

Reply:
Also it would be good to memoize the file (like the Brexit checker) so it remain in memory. We don't have to load the file for every request.

Reply:
Hi @theseanything - thanks for the review and your comments.  I'll definitely be looking into memoizing the file as you suggest and I agree with your suggested change to the `unavailable?` method.

However, I don't believe it does represent a single service - no.  It's more like - all local transaction services that are unavailable in the named countries as defined in the [config](config/unavailable_services.yml) file.  I understand the current name is a bit confusing, and your request below to rename it below.  I agree there must be a better name.  How would you feel about `UnavailableLocationTransactionServices` - bit of a mouthful, but is definitely more illustrative of my intent, its use and purpose?

My reasoning for encapsulating the config into an object like this, is that I wanted change or add the minimal amount of code - especially in the `LocalTransactionController`.

Reply:
Actually, the more I think about this class and your suggestion to memoize the file - the more this looks like a `singleton`.

Reply:
Class `LocalTransactionService` is now a Singleton.  Config file loaded only once.